### PR TITLE
SW BRC fixes for 19.1

### PR DESCRIPTION
**Pull request by aulitin**

### CODE REVIEW

File: _studio/mfx_lib/shared/include/mfx_brc_common.h [hunk 1]
```diff
@@ -42,8 +42,8 @@ class AVGBitrate
         m_maxWinBits(maxBitPerFrame*windowSize),
         m_maxWinBitsLim(0),
         m_avgBitPerFrame(MFX_MIN(avgBitPerFrame, maxBitPerFrame)),
-        m_currPosInWindow(0),
-        m_lastFrameOrder(0),
+        m_currPosInWindow(windowSize-1),
```
Comment (dvrogozh) [inline _studio/mfx_lib/shared/include/mfx_brc_common.h]: what if windowSize=0?

File: samples/sample_common/include/brc_routines.h [hunk 2]
```diff
@@ -211,8 +211,8 @@ class AVGBitrate
         m_maxWinBits(maxBitPerFrame*windowSize),
         m_maxWinBitsLim(0),
         m_avgBitPerFrame(IPP_MIN(avgBitPerFrame, maxBitPerFrame)),
-        m_currPosInWindow(0),
-        m_lastFrameOrder(0)
+        m_currPosInWindow(windowSize-1),
```
Comment (dvrogozh) [inline samples/sample_common/include/brc_routines.h]: what is windowSize=0?

**Comment by dvrogozh** [comment]
"Validation" is not the best name for pull request.

**Comment by ogolikov** [comment]
"what is windowSize=0?"
m_currPosInWindow = mfxU32(-1);
the first call:
m_currPosInWindow = (m_currPosInWindow + 1) % windowSize; //window size =1
so m_currPosInWindow = 0

**Pull request merged**